map_model/src/make/turns.rs

```diff
@@ -474,7 +474,7 @@ fn make_degenerate_crosswalks(
     lanes: &Vec<Lane>,
     r1: &Road,
     r2: &Road,
-) -> Option<Vec<Turn>> {
+) -> Option<impl Iterator<Item=Turn>> {
```

Comment:
I'm curious about the motivation behind this kind of change. To me, `Vec<Turn>` is more readable. `impl Iterator` is more precise, because the fact the result is an allocated vector doesn't really matter. But this is used in just one place, and it's not on a critical path at all, so performance isn't an issue. Is it preferable generally to express these situations as iterables? I haven't taken much time to read the Rust style guides floating around, so I'm curious about best practices

Reply:
I tend to agree that `Vec` is more readable, but `make_degenerate_crosswalks` is only called in a context where we essentially directly iterate over the elements returned by it. Thus, we do not need to do the work to `collect` them into a temporary data structure.